compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/test/fuse_and_hoist_forall.mlir

```diff
@@ -265,4 +215,48 @@ func.func @multi_hoist_and_fuse_trailing_stuff() {
 //       CHECK:       linalg.ceil ins(%[[T]] : tensor<4x2xf16>) {{.*}} -> tensor<4x2xf16>
 //       CHECK:     scf.forall.in_parallel
 //       CHECK:   scf.forall.in_parallel
-//       CHECK:   flow.dispatch.tensor.store
+//       CHECK:   return
+
+// -----
+
+func.func @multi_hoist_and_fuse_trailing_with_producer_fusion(%2: tensor<128x128xf16>, %3: tensor<128x128xf16>) -> tensor<128x128xf16> {
```

Comment:
I'm curious how a case like this would come up?

Reply:
This happens from packing. For example
```
%mma = iree_gpu.multi_mma
%unpack = tensor.unpack
linalg.add %unpack, %1
```
after pack propagation becomes
```
%mma = iree_gpu.multi_mma
%pack = tensor.pack %1
%add = linalg.add %mma, %pack
tensor.unpack %add
```
so we end up with a pack on one of the other inputs (this happens basically 100% of the time with fusion of, say, bias adds).